mir_eval/separation.py

```diff
@@ -300,6 +325,141 @@ def bss_eval_sources_framewise(reference_sources, estimated_sources,
     return sdr, sir, sar, perm
 
 
+def bss_eval_images(reference_sources, estimated_sources,
+                    compute_permutation=True):
+    """Implementation of the bss_eval_images function from the
+    BSS_EVAL Matlab toolbox.
+
+    Ordering and measurement of the separation quality for estimated source
+    signals in terms of filtered true source, interference and artifacts.
+    This method also provides the ISR measure.
+
+    The decomposition allows a time-invariant filter distortion of length
+    512, as described in Section III.B of [#vincent2006performance]_.
+
+    Passing True for compute_permutation will much improve the performance of
+    the evaluation; however, it is not always appropriate and is not the
+    way that the BSS_EVAL Matlab toolbox computes bss_eval_images.
+
+    Examples
+    --------
+    >>> # reference_sources[n] should be an ndarray of samples of the
+    >>> # n'th reference source
+    >>> # estimated_sources[n] should be the same for the n'th estimated
+    >>> # source
+    >>> (sdr, isr, sir, sar,
+    ...  perm) = mir_eval.separation.bss_eval_images(reference_sources,
+    ...                                               estimated_sources)
+
+    Parameters
+    ----------
+    reference_sources : np.ndarray, shape=(nsrc, nsampl, nchan)
+        matrix containing true sources
+    estimated_sources : np.ndarray, shape=(nsrc, nsampl, nchan)
+        matrix containing estimated sources
+    compute_permutation : bool, optional
+        compute permutation of estimate/source combinations (True by default)
+
+    Returns
+    -------
+    sdr : np.ndarray, shape=(nsrc,)
+        vector of Signal to Distortion Ratios (SDR)
+    isr : np.ndarray, shape=(nsrc,)
+        vector of source Image to Spatial distortion Ratios (ISR)
+    sir : np.ndarray, shape=(nsrc,)
+        vector of Source to Interference Ratios (SIR)
+    sar : np.ndarray, shape=(nsrc,)
+        vector of Sources to Artifacts Ratios (SAR)
+    perm : np.ndarray, shape=(nsrc,)
+        vector containing the best ordering of estimated sources in
+        the mean SIR sense (estimated source number perm[j] corresponds to
+        true source number j)
+        Note: perm will be (1,2,...,nsrc) if compute_permutation is False
+
+    """
+
+    # make sure the input has 3 dimensions
```

Comment:
You should be able to replace all of the following code (up to line 391) with a single call to `np.atleast_3d`.